### ISSUE TITLE
Avoid crossing perimeter (is still crossing perimeter)

### ISSUE THREAD

**Issue by TommyC81** [supermerill/SuperSlicer#2398]
### What happened?

I'm using setting "Avoid crossing perimeter" with "Max detour length" set to 0 (zero) - this should disable the detour length consideration and simply avoid crossing the perimeter all the time. However, I'm getting lots of perimeter crossing in places where I'd expect it to be avoided.

Example project and images attached.

### Project file & How to reproduce

[Avoid perimeter crossing - example.zip](https://github.com/supermerill/SuperSlicer/files/8077176/Avoid.perimeter.crossing.-.example.zip)
![Avoid perimeter crossing - setting](https://user-images.githubusercontent.com/1904092/154205803-dc8c9a77-0728-414c-8ff2-8e4aa8a3b052.PNG)
![Avoid perimeter crossing - example](https://user-images.githubusercontent.com/1904092/154205813-78dd439f-bc3a-47a7-9fa2-7601ea51ba8d.PNG)

### Version

2.3.57.10

### Operating system

Windows 10

### Printer model

Creality CR6

**Comment by Pavulon87** [comment]
It looks like this has stopped working since SuperSlicer_2.3.57.7
2.3.56.6:
![2 3 57 6](https://user-images.githubusercontent.com/23641103/154225372-c763fae5-d533-46c0-a467-aa7cd556d1fa.jpg)
2.3.57.7 and newer:
![2 3 57 7](https://user-images.githubusercontent.com/23641103/154225400-7d7e0b9a-6008-4c6e-8b91-d21d430d3af0.jpg)

**Comment by TommyC81** [comment]
![2022-02-19 - Nightly avoid perimeter fail](https://user-images.githubusercontent.com/1904092/154834985-8608c9f1-77e5-4b16-8983-05565afdbd0f.PNG)
[Avoid perimeter crossing - example (nightly 2022-02-19).zip](https://github.com/supermerill/SuperSlicer/files/8103648/Avoid.perimeter.crossing.-.example.nightly.2022-02-19.zip)

To confirm, the avoid crossing perimeter works better in the nightly from 2022-02-19. However, note the repeated perimeter crossing that I highlighted in previous post is still occurring. Image and project file attached.

**Comment by supermerill** [comment]
The new algo from 1.3.57.7 need a bit more width (#1794), hence why it's now failing at this object ( it needs to be scaled by 140% to recover good behavior).

The algo should fallback to 'old' behavior of following the external periemter whan it's too thin, I'll have to investigate a lot to find why it's failing.

**Comment by Ygle38** [comment]
Hello, 
Just another case of crossing perimeter, with supports.
- this part is ok with no crossing (in case of no support), but adding support reveal the failure; travelling from left to right support by crossing the main part.
- probably the same issue, I post it in case it helps to understand the code regression
![Capture_vtol_crossing](https://user-images.githubusercontent.com/103949112/167690920-ff17a559-8ff0-4151-97ed-7965d47bf336.JPG)
[Vtol1.zip](https://github.com/supermerill/SuperSlicer/files/8663847/Vtol1.zip)

**Comment by c2v4** [comment]
Yeah, I have the same issue. I found out that this only shows up for a certain range of travel:
![image](https://user-images.githubusercontent.com/11049808/168752393-1afa110c-7aff-425f-8d34-711fa
[k40exhaust-Body1-1.zip](https://github.com/supermerill/SuperSlicer/files/8706093/k40exhaust-Body1-1.zip)
69b0fb0.png)
In layers that are close to those that are affected by it, infill join is in the same place as well as the 'turn point' of internal perimeter. I've attached my project file in case that would help you find the answer. SS Version: 2.4.58.2
